Gcc10 settable flags

kgerheiser opened the pull request:
Also remove `-free` flags from the compiler and change the file extension to `.f90` instead

kgerheiser committed:
GCC 10 flags, compiler check, and settable flags

Separate C and Fortran flags

kgerheiser committed:
Change .f with -free compiler flags to .f90

kgerheiser committed:
Lowercase filenames

climbfuji reviewed (approved):
Ok by me. I usually prefer to use uppercase filename extensions to enable automatic preprocessing when needed.

kgerheiser committed:
Rename baciof.f90 to baciof.F90

kgerheiser committed:
Rename bafrio.f90 to bafrio.F90

kgerheiser committed:
Rename chk_endianc.f90 to chk_endianc.F90

aerorahul commented on src/CMakeLists.txt:
```diff
@@ -7,16 +7,28 @@ elseif(UNIX)
 if(CMAKE_C_COMPILER_ID MATCHES "^(Intel)$")
-  set(CMAKE_Fortran_FLAGS "-g -fpp -free -xHOST -traceback -FR")
-  set(fortran_8_flags "-i8 -r8")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
```
Don't think this is necessary.

kgerheiser replied:
Right, I see that now. If nothing overwrites the flag it's fine.

aerorahul commented on src/CMakeLists.txt:
```diff
@@ -7,16 +7,28 @@ elseif(UNIX)
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
+  set(CMAKE_C_FLAGS_DEBUG "-g -traceback")
 elseif(CMAKE_C_COMPILER_ID MATCHES "^(GNU|Clang|AppleClang)$")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
```
Not necessary.

kgerheiser committed:
.f90 to .F90 in list of files

kgerheiser commented:
That's what I was going to do, but then I already named the `.f90` on my Mac, which has a case-insensensitive filesystem and git wouldn't recognize the change. So, I changed it in CMakeLists.

But anyway, I went back and changed them through the Github GUI

aerorahul commented on src/CMakeLists.txt:
```diff
@@ -7,16 +7,28 @@ elseif(UNIX)
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -w -fallow-argument-mismatch")
+endif()
+
+set(fortran_src baciof.F90 bafrio.F90 chk_endianc.F90)
```
Are there preprocessor directives in these?

kgerheiser replied:
No

aerorahul replied:
I think only `bacio.F90` has `include` and that is fine.
The other 2 can be `.f90`

climbfuji replied:
We probably have diverging philosophies on this. My take is that preprocessing the file doesn't cost much and avoids potential changes in the future. But either way is ok for me.

aerorahul replied:
- Just looking at the file extension tells us that there is preprocessor involved.
- CMake processes those files first to produce `.f90` which are then compiled.

kgerheiser replied:
That's actually a Fortran `Include` which doesn't use the pre-processor

aerorahul replied:
even better. remove the `-fpp` completely and turn those files to `.f90`

aerorahul reviewed (changes requested):
Empty CMAKE_C_FLAGS don't need to be set.  They will be taken directly from env.
See comment on preprocessor files for `.F90`

kgerheiser committed:
Remove empty CMAKE_C_FLAGS

aerorahul commented on src/CMakeLists.txt:
```diff
@@ -7,16 +7,26 @@ elseif(UNIX)
 if(CMAKE_C_COMPILER_ID MATCHES "^(Intel)$")
-  set(CMAKE_Fortran_FLAGS "-g -fpp -free -xHOST -traceback -FR")
-  set(fortran_8_flags "-i8 -r8")
+  set(CMAKE_C_FLAGS_DEBUG "-g -traceback")
```
Actually, this should be `set(CMAKE_C_FLAGS "-g -traceback ${CMAKE_C_FLAGS}")`

kgerheiser committed:
Fix flags

kgerheiser committed:
Rename baciof.F90 to baciof.f90

kgerheiser committed:
Rename bafrio.F90 to bafrio.f90

kgerheiser committed:
Rename chk_endianc.F90 to chk_endianc.f90

aerorahul commented on src/CMakeLists.txt:
```diff
@@ -7,16 +7,28 @@ elseif(UNIX)
+endif()
+
+if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
+  set(CMAKE_Fortran_FLAGS "-g -fpp -traceback -FR ${CMAKE_Fortran_FLAGS}")
```
i think you missed a spot `-fpp`

kgerheiser committed:
Remove -fpp
